reminder/module.py

```diff
@@ -0,0 +1,465 @@
+                    status=ReminderStatus.str_list()
+                )
+            )
+            return
+
+        query = ReminderItem.get_all(guild_id=ctx.guild.id, status=status)
+        await self._send_reminder_list(ctx, query)
+
+    @commands.guild_only()
+    @commands.check(check.acl)
+    @reminders_.command(pass_context=True, aliases=["postpone", "delay"])
+    async def reschedule(self, ctx, idx: int, datetime_str: str):
+        """Reschedule reminder"""
+        query = ReminderItem.get_all(guild_id=ctx.guild.id, idx=idx)
+        if query is None:
+            await ctx.send(
+                _(ctx, "Reminder with ID {id} does not exists.").format(id=idx)
+            )
+            return
+
+        query = query[0]
+
+        if query.remind_id != ctx.author.id:
+            await ctx.send(_(ctx, "Can't reschedule other's reminders."))
+            return
+
+        try:
+            date = utils.time.parse_datetime(datetime_str)
+        except dateutil.parser.ParserError:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "I don't know how to parse `{datetime_str}`, please try again.",
+                ).format(datetime_str=datetime_str)
+            )
+            return
+
+        if date < datetime.now():
+            await ctx.send(_(ctx, "Reschedule time must be in furuter."))
+            return
+
+        print_date = date.strftime("%d.%m.%Y %H:%M")
+
+        embed = await self._get_embed(ctx, query)
+        embed.add_field(
+            name=_(ctx, "New time"),
+            value=print_date,
+            inline=False,
+        )
+        embed.add_field(
+            name=_(ctx, "Do you really want to edit reminder time/date?"),
+            value=_(ctx, "✅ for confirmation") + _(ctx, "❎ for storno"),
+            inline=False,
+        )
+
+        message = await ctx.send(embed=embed)
+
+        await message.add_reaction("✅")
+        await message.add_reaction("❎")
+
+        def check_id(reaction, user_id):
+            return (
+                reaction.message.id == message.id
+                and (str(reaction.emoji) == "✅" or str(reaction.emoji) == "❎")
+                and user_id.id == query.remind_id
+            )
+
+        try:
+            reaction, user = await self.bot.wait_for(
+                "reaction_add", check=check_id, timeout=60
+            )
+        except asyncio.TimeoutError:
+            await ctx.send(_(ctx, "Reschedule timed out."))
+        else:
+            if str(reaction.emoji) == "✅":
+                await bot_log.debug(
+                    ctx.author,
+                    ctx.channel,
+                    f"Rescheduling reminder - ID: {query.idx}, time: {date}, status: {query.status}",
+                )
+
+                query.reschedule(date)
+                ctx.send(_(ctx, "Reminder rescheduled."))
+            elif str(reaction.emoji) == "❎":
+                ctx.send(_(ctx, "Rescheduling aborted."))
```

Comment:
We should try to make this into a utils function. Not necessary now, just pointing out for the future.

Possibly while using the new Discord Components (like ActionRows and Buttons)

Reply:
I was talking about this with sinus-x and you are right, this should be in core.